### DISCUSSION TITLE
Blue Rectangle - 2D platformer using Dunai and Bearriver

### DISCUSSION THREAD

**Discussion by AntanasKal** [ivanperez-keera/dunai#379] [Show and tell]
Hello,

I have created a demo platformer [Blue Rectangle](https://github.com/AntanasKal/blue-rectangle) using Dunai and Bearriver libraries and SDL2 bindings. The game implements continuous collision detection as outlined in the [Functional Reactive Programming, Refactored](https://www.cs.nott.ac.uk/~psxip1/papers/2016-HaskellSymposium-Perez-Barenz-Nilsson-FRPRefactored-short.pdf) paper. 

In the paper a 1D bouncing ball example is given, but I was interested to see whether this technique can be applied to slightly more complicated examples (I was not able to find such examples yet). I also wanted to implement simple 2D platformer physics in a novel way. In the game some significant simplifications are still applied, such as: accelerations are constant (so the object trajectories are piecewise parabolas), no rotations are allowed, object shapes are rectangular. These greatly simplify physics and collision time prediction calculations. Also, I have not spent time optimizing the performance.

![gameplay](https://github.com/ivanperez-keera/dunai/assets/6152388/667c8129-d3c2-4d31-8c52-afce64a317fa)

Hopefully that is interesting to you. Having warmed up with Dunai and Bearriver, I would be interested in contributing to the libraries :)

Antanas

**Comment by ivanperez-keera** [comment]
OMG this is awesome!!! 🎉🎉🎉

**Comment by ivanperez-keera** [comment]
@AntanasKal would it be cool if I added this video to the announcement of the next version of dunai and bearriver?

**Reply by AntanasKal** [reply]
Yes, feel free to add it :)

**Reply by ivanperez-keera** [reply]
Thanks! :)
